### PR TITLE
Indirect assignment of cvar values

### PR DESCRIPTION
**Pull request by alexeykostin**
## What does this PR do?

Indirect assignment of cvar values to keep the assigned values in sync with the legacy console.

## How was this PR tested?

Tested locally

### CODE REVIEW

File: Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl [hunk 1]
```diff
@@ -37,7 +37,13 @@ namespace AZ
         // Client code can supply a type for m_value that overrides the operator= function and trigger side effects
         // in the operator= function body. Doing the assignment outside the value change check avoids those side
         // effects not being triggered because AzCore believes the value wouldn't change.
+
+#if !defined(_RELEASE) && defined(CARBONATED)
```
Comment (aefimov3) [inline Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl]: I suggest Release to function same as non-release, the less differences we have the better. Please analyze the direct cvar assignment frequency.

File: Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.cpp [hunk 2]
```diff
@@ -172,4 +172,14 @@ namespace AZ
     {
         return GetValueResult::NotImplemented;
     }
+
+#if !defined(_RELEASE) && defined(CARBONATED)
+    void ConsoleFunctorBase::PerformCommand(CVarFixedString inString)
+    {
+        if (m_console)
```
Comment (aefimov3) [inline Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.cpp]: Can it be that cvars being assigned without console? Please add "else", then set a breakpoint to check. Maybe two platforms PC & iOS.

Comment (alexeykostin) [reply Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.cpp]: I have not seen such cases.

Technically, `m_console` can be nullptr in `ConsoleFunctorBase`. So checks and a fallback when the console is not available added.

File: Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl (hunk 1 above)
Comment (aefimov3) [reply Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl]: Let wait for Lloyd's feedback on RELEASE condition.

Comment (Twolewis) [reply Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl]: I agree w/ @aefimov3 here.  We want consistent behaviour in all configurations.

**Review by Twolewis — CHANGES_REQUESTED** [review]
request to remove RELEASE guard

File: Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl (hunk 1 above)
Comment (alexeykostin) [reply Code/Framework/AzCore/AzCore/Console/ConsoleDataWrapper.inl]: Removed

**Review by aefimov3 — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.cpp [hunk 3]
```diff
@@ -172,4 +172,22 @@ namespace AZ
...
+    {
+        auto console = m_console;
+        if (!console)
+        {
+            console = AZ::Interface<IConsole>::Get();
+        }
+        if (console)
+        {
+            auto res =
+                m_console->PerformCommand(AZStd::string::format("%s %s", m_name, inString.c_str()).c_str(), ConsoleSilentMode::Silent);
```
Comment (carbonated-niko) [inline Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.cpp]: Should this line be console->Perform command? That first if implies m_console could be null, unless getting the iconsole sets that somehow as a sideeffect

Comment (alexeykostin) [reply Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.cpp]: Fixed

**Review by carbonated-niko — APPROVED** [review]

**Review by Twolewis — APPROVED** [review]

**Pull request merged**